celest/schedule/insertion_operators.py

```diff
@@ -0,0 +1,128 @@
+
+
+from celest.units.quantity import Quantity
+from celest import units as u
+import math
+import numpy as np
+
+
+def greedy_insertion(request_handler, number_to_insert):
+    request_handler.sort_by_decreasing_priority()
+    insert_first_n_requests(request_handler, number_to_insert)
+
+
+def insert_first_n_requests(request_handler, number_to_insert):
+    for request_index in range(request_handler.number_of_requests):
+
+        if not number_to_insert:
+            break
+
+        if request_handler.is_request_scheduled(request_index):
+            continue
+
+        deadline = request_handler.deadline(request_index)
+        duration = request_handler.duration(request_index)
+        minimum_image_quality = request_handler.image_quality(request_index)
+        look_angle = request_handler.look_angle(request_index)
+        request_vtw_list = request_handler.vtw_list(request_index)
+
+        duration_in_days = Quantity(duration.to(u.s) / 86400, u.jd2000)
+
+        for vtw_index, vtw in enumerate(request_vtw_list):
```

Comment:
Some comments in this for loop or extracting functions would provide necessary clarity.